openlayers/openlayers.d.ts

```diff
@@ -203,7 +203,7 @@ declare module ol {
          * @template S
          * @api stable
          */
-        forEach<S>(f: ((self: S, item: T, index: number, array: T[]) => any), opt_this?: S): void;
+        forEach<S, T>(f: ((item: T, index: number, array: T[]) => any), opt_this?: S): void;
```

Comment:
Agreed.  I wasn't sure about this when I made the change and didn't notice definition of T in the class declaration.  Thanks for the clarification.  There is now an additional 4th commit in this PR changing it back to how it was.  

Also, thanks for your work on this.  The definitions you have created in the last week or so are far more complete than those previously available.